### DISCUSSION TITLE
using Auxvaribles from previous step for extra-stress calculation

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#17408] [Q&A Modules: Solid mechanics] [answered]
Hi guys,

I defined Auxvaribale as below, I just wonder how can I use it for extra-stress calculation? seems like I cannot just call the variable name.

```
[AuxVariables]
  [./stress_xx]
    order = SECOND
    family = MONOMIAL
  [../]
  [./stress_yy]
    order = SECOND
    family = MONOMIAL
  [../]
  [./stress_zz]
    order = SECOND
    family = MONOMIAL
  [../]
[]
[AuxKernels]
  [./matl_s11]
    type = RankTwoAux
    rank_two_tensor = stress
    index_i = 0
    index_j = 0
    variable = stress_xx
  [../]
  [./matl_s22]
    type = RankTwoAux
    rank_two_tensor = stress
    index_i = 1
    index_j = 1
    variable = stress_yy
  [../]
  [./matl_s33]
    type = RankTwoAux
    rank_two_tensor = stress
    index_i = 2
    index_j = 2
    variable = stress_zz
  [../]
[]
```
below is for the extra-stress
```
[./void_stress]
  type = ComputeExtraStressConstant
  block = 1
  extra_stress_tensor = '-stress_xx -stress_yy -stress_zz 0 0 0'
  prefactor = Ex_stress1
[../]

[./Ex_stress1]
  type = ParsedFunction
  value = 'if(t<2,1,1e-4)'
[../]
```
here is the error message: `cannot convert '-stress_zz' to float`

It should be quite straightforward, but I couldn't find the way around it.

Thank you!
Traiwit

**Comment by Traiwit** [comment]
note that: the objective here is to remove the stress for a particular block, hence, extra_stress_tensor equal to previous step stress but in opposite direction

**Comment by aeslaughter** [comment, marked as answer]
You should be able to setup the stress to only be calculated on the desired block and setup another on the other block that is zero, since the stress calculations are Material objects.

But, I am not familiar with mechanics module; perhaps some mechanics folks could jump in @jiangwen84 or @sapitts with the correct approach.

**Reply by jiangwen84** [reply]
The name of `ComputeExtraStressConstant` indicates that it only works for a constant stress tensor. If the stress in your simulation is not a constant, i.e., with spatial variation, you cannot use this object.

Could you explain a bit more on what you are trying to do? Do you want zero stress in the void?

**Reply by Traiwit** [reply]
Hi, yes, I instead modified the computestress .C file by adding the prefactor term, so now I can multiply the old stress by some prefactor which can reduce the stress in the particular block (creating the void). 

All good now, thanks guys!

```
{

  _stress[_qp] = _stress_old[_qp]* _prefactor[_qp]+ _elasticity_tensor[_qp] * elastic_strain_increment;

  computeQpJacobian();
}
```
